code/modules/antagonists/eldritch_cult/knowledge.dm/ash_lore.dm

```diff
@@ -0,0 +1,202 @@
+/datum/eldritch_knowledge/base_ash
+	name = "Nightwatcher's secret"
+	desc = "Opens up the path of ash to you. Allows you to transmute a match with a knife into an ashen blade."
+	gain_text = "City guard knows their watch. If you ask them at night they may tell you about the ashy lantern."
+	banned_knowledge = list(/datum/eldritch_knowledge/base_rust,/datum/eldritch_knowledge/base_flesh,/datum/eldritch_knowledge/rust_final,/datum/eldritch_knowledge/flesh_final)
+	next_knowledge = list(/datum/eldritch_knowledge/ashen_grasp)
+	required_atoms = list(/obj/item/kitchen/knife,/obj/item/match)
+	result_atoms = list(/obj/item/melee/sickly_blade/rust)
+	cost = 1
+	route = "Ash"
+
+/datum/eldritch_knowledge/spell/ashen_shift
+	name = "Ashen Shift"
+	gain_text = "Ash is all the same, how can one man master it all?"
+	desc = "Short range jaunt that can help you escape from bad situations."
+	cost = 1
+	spell_to_add = /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash
+	next_knowledge = list(/datum/eldritch_knowledge/ash_mark,/datum/eldritch_knowledge/essence,/datum/eldritch_knowledge/ashen_eyes)
+	route = "Ash"
+
+/datum/eldritch_knowledge/ashen_grasp
+	name = "Ashen Grasp"
+	gain_text = "Gates have opened, minds have flooded, I remain."
+	desc = "Empowers your mansus grasp to throw away enemies."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/spell/ashen_shift)
+	route = "Ash"
+
+/datum/eldritch_knowledge/ashen_grasp/mansus_grasp_act(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!iscarbon(target))
+		return
+
+	var/mob/living/carbon/C = target
+	var/datum/status_effect/eldritch/E = C.has_status_effect(/datum/status_effect/eldritch/rust) || C.has_status_effect(/datum/status_effect/eldritch/ash) || C.has_status_effect(/datum/status_effect/eldritch/flesh)
+	if(E)
+		E.on_effect()
+		for(var/X in user.mind.spell_list)
+			if(!istype(X,/obj/effect/proc_holder/spell/targeted/touch/mansus_grasp))
+				continue
+			var/obj/effect/proc_holder/spell/targeted/touch/mansus_grasp/MG = X
+			MG.charge_counter = min(round(MG.charge_counter + MG.charge_max * 0.75),MG.charge_max) // refunds 75% of charge.
+	var/atom/throw_target = get_edge_target_turf(C, user.dir)
+	if(!C.anchored)
+		C.throw_at(throw_target, rand(4,8), 14, user)
+	return
+
+/datum/eldritch_knowledge/ashen_eyes
+	name = "Ashen Eyes"
+	gain_text = "Piercing eyes may guide me through the mundane."
+	desc = "Allows you to craft thermal vision amulet by transmutating eyes with a glass shard."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/spell/ashen_shift,/datum/eldritch_knowledge/flesh_ghoul)
```

Comment:
aren't flesh ghouls supposed to be a flesh thing?

Reply:
yeah path of exile style skill tree, you can do that.